go-controller/pkg/node/default_node_network_controller.go

```diff
@@ -342,19 +342,27 @@ func setupOVNNode(node *kapi.Node) error {
 	return nil
 }
 
-func setEncapPort() error {
+func setEncapPort(ctx context.Context) error {
 	systemID, err := util.GetNodeChassisID()
 	if err != nil {
 		return err
 	}
-	uuid, _, err := util.RunOVNSbctl("--data=bare", "--no-heading", "--columns=_uuid", "find", "Encap",
-		fmt.Sprintf("chassis_name=%s", systemID))
+	var uuid string
+	err = wait.PollUntilContextTimeout(ctx, 250*time.Millisecond, 30*time.Second, true,
+		func(ctx context.Context) (bool, error) {
+			uuid, _, err = util.RunOVNSbctl("--data=bare", "--no-heading", "--columns=_uuid", "find", "Encap",
+				fmt.Sprintf("chassis_name=%s", systemID))
+			if err != nil {
+				return false, err
+			}
+			if len(uuid) == 0 {
+				return false, fmt.Errorf("unable to find encap uuid to set geneve port for chassis %s", systemID)
```

Comment:
I think the whole retry will stop on error, right? So do we want to return errors here at all?

Reply:
@npinaeva yes, you are correct. now not returning error in the retry loop when uuid is empty.